server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
```

Comment:
this makes sense if we decide to respond with non-200/500 HTTP responses (exception can be thrown to say 500) in some cases.
We can send 400 Bad Request, if someone asks for a data source which is not present but that will be inconsistent with rest of the broker. Usually, I noticed, broker responds successfully with "empty" response in most of such cases.